lib/Sema/TypeCheckConcurrency.cpp

```diff
@@ -5247,6 +5246,22 @@ AnyFunctionType *swift::adjustFunctionTypeForConcurrency(
   // Apply unsafe concurrency features to the given function type.
   bool strictChecking = contextRequiresStrictConcurrencyChecking(
       dc, getType, isolatedByPreconcurrency);
+
+  auto &ctx = dc->getASTContext();
+  bool inferredSendable =
+      ctx.LangOpts.hasFeature(Feature::InferSendableMethods);
+
+  if (inferredSendable) {
+    if (auto func = dyn_cast_or_null<AbstractFunctionDecl>(decl)) {
+      auto doesNotCapture = func->getCaptureInfo().getCaptures().empty();
+
+      if (doesNotCapture) {
+        fnType = fnType->withExtInfo(fnType->getExtInfo().withConcurrent());
```

Comment:
I don't understand why this is necessary. Is there a specific test case that doesn't work without this change?

Reply:
I was thinking this might be a good way to detect if a function is global, because global functions don't capture anything .

Reply:
The global function test cases aren't handled by the ConstraintSystem checks

Reply:
I think the rule is that local functions are never `@Sendable` unless explicitly annotated, which means we don't need to check the captures inside of `ConstraintSystem::getTypeOfReference`. We can just check that the function's DeclContext is not `isLocalContext()` before applying the `@Sendable` attribute.

Reply:
oh didn't know about isLocalContext! Updated.